### ISSUE TITLE
Suggestion: Stable Diffusion Model

### ISSUE THREAD

**Issue by jafri** [ggerganov/ggml#4]
Been following along with your speed increases on Whisper using ggml, which have been amazing

Would be interesting to see how stable diffusion runs on CPUs using ggml

Here are current benchmarks:
![image](https://user-images.githubusercontent.com/13677357/197418608-9e43af3e-95a3-4ccb-82a1-c5c565b7d892.png)

Popular way to run SD:
https://github.com/huggingface/diffusers

**Comment by ggerganov** [comment]
Yup, I've been thinking about it - the network seems a bit more complex compared to GPT and Whisper, but I think most of the building blocks are already available in `ggml`.

Last time I took a look at the python source, I got a bit lost because the model does not seem concentrated in one place as it is with the other models. For example, [here](https://github.com/huggingface/diffusers/tree/main/src/diffusers/pipelines) I see 8 pipelines and I don't know which one is relevant and which not.

**Comment by jafri** [comment]
This one is interesting too:

https://github.com/TheLastBen/fast-stable-diffusion

**Comment by x-legion** [comment]
> Yup, I've been thinking about it - the network seems a bit more complex compared to GPT and Whisper, but I think most of the building blocks are already available in `ggml`.
> 
> Last time I took a look at the python source, I got a bit lost because the model does not seem concentrated in one place as it is with the other models. For example, [here](https://github.com/huggingface/diffusers/tree/main/src/diffusers/pipelines) I see 8 pipelines and I don't know which one is relevant and which not.

![IMG_20230428_143440.jpg](https://user-images.githubusercontent.com/97628039/235098102-ad95461c-d2bc-4136-8ec3-2b7c092a88c7.jpg)

**Comment by tikikun** [comment]
hi thumb up for this

**Comment by leejet** [comment]
I've implemented Stable Diffusion using GGML. Welcome to try it out.

https://github.com/leejet/stable-diffusion.cpp

This is an example

```
./sd -m ../models/sd-v1-4-ggml-model-f16.bin -p "a lovely cat"
```

![image](https://github.com/ggerganov/ggml/assets/31925346/55f5c61f-11ba-4639-aacc-c73c7e8fddfa)
